runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldTableRow.java

```diff
@@ -326,26 +313,12 @@ private JPanel buildWorldField()
 
 		worldField = new JLabel(world.getId() + "");
 
-		JLabel flag = new JLabel(getFlag(world.getLocation()));
+		JLabel flag = new JLabel(new ImageIcon(ImageUtil.getResourceStreamFromClass(WorldHopperPlugin.class,
+			"flag_" + world.getLocation().getTwoLetterCode().toLowerCase() + ".png")));

```

Comment:
This change would duplicate the icon/image initialization per table row generated. Instead, cache the icons in a `static` block (like was done before) using a `Map<WorldRegion, ImageIcon>`.